### PR TITLE
Marines wraith

### PR DESCRIPTION
**Pull request by PabstMirror**
Replace #62 

- [x] Redo markers defines to current file format
- [x] Finish the mission.sqm and put it in
- [x] Make sure there's 4 IFV's and 4 tanks because reasons.
- [x] Add incldues in cfgloadouts
- [x] Add it to respawn

### CODE REVIEW

**Comment by Alex6511** [comment]
@PabstMirror You're probably going to need to cleanup the woodland file too.

**Comment by PabstMirror** [comment]
woodland is good

**Comment by Alex6511** [comment]
Right then, this should be ready to merge then.

File: CfgLoadouts.hpp [hunk 1]
```diff
@@ -23,6 +23,10 @@ class CfgLoadouts {
   // #include "Loadouts\blu_swede_ak5_m90.hpp"     // Swedish: AK5 - M90 (Green)
   // #include "Loadouts\blu_ger_g36_fleck.hpp"     // German: G36 - Flecktarn Camo
 
+  //USMC FACTION (rhs_faction_usmc_wd):
+  #include "Loadouts\blu_usmc_m16_desert.hpp"      // USMC COOP Template Desert Loadout
```
Comment (AACO) [inline CfgLoadouts.hpp]: See comment above for standardized format

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+  #define WEST_RIFLE_MAG "rhs_mag_30Rnd_556x45_Mk318_Stanag:8","rhs_mag_30Rnd_556x45_M855A1_Stanag_Tracer_Red:2"
+  #define WEST_RIFLE_ATTACHMENTS "rhsusf_acc_anpeq15A"
+  #define WEST_AAR_ATTACHMENTS WEST_RIFLE_ATTACHMENTS
+  //GL Rifle
+  #define WEST_GLRIFLE "rhs_weap_m16a4_carryhandle_M203"
+  #define WEST_GLRIFLE_MAG WEST_RIFLE_MAG
+  #define WEST_GLRIFLE_MAG_SMOKE "rhs_mag_m714_White:2","rhs_mag_m713_Red:2"
+  #define WEST_GLRIFLE_MAG_HE "rhs_mag_M433_HEDP:5"
+  #define WEST_GLRIFLE_MAG_FLARE "rhs_mag_M585_white:2","rhs_mag_m661_green:2","rhs_mag_m662_red:2"
+
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: Remove empty line

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  #define WEST_GLLAUNCHER_MAG_HE "rhsusf_mag_6Rnd_M441_HE:3"
+  #define WEST_GLLAUNCHER_MAG_SMOKE "rhsusf_mag_6Rnd_M714_white:5"
+  //Carbine
+  #define WEST_CARBINE WEST_RIFLE
+  #define WEST_CARBINE_MAG WEST_RIFLE_MAG
+  // AR
+  #define WEST_AR "rhs_weap_m249_pip_L"
+  #define WEST_AR_MAG "rhs_200rnd_556x45_M_SAW:2"
+  // AT
+  #define WEST_AT "rhs_weap_rpg26"
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: RPG 26?

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+  #define WEST_AT_MAG "rhs_rpg26_mag"
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: RPG 26 doesn't need a mag entry

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+  // MAT
+  #define USMC_MAT "rhs_weap_smaw_green"
+  #define USMC_MAT_MAG "rhs_mag_smaw_HEDP:2", "rhs_mag_smaw_SR:5"
+  #define USMC_MAT_MAG2 "rhs_mag_smaw_HEAA:2"
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: The gunner can hold an extra mag. Also consider giving MAT gunner and spotter mixed loads.

File: Loadouts/blu_usmc_m16_desert.hpp (hunk 3 above)
Comment (Alex6511) [inline Loadouts/blu_usmc_m16_desert.hpp]: https://en.wikipedia.org/wiki/M72_LAW

File: CfgLoadouts.hpp (hunk 1 above)
Comment (Alex6511) [inline CfgLoadouts.hpp]: Don't understand what you mean, I followed the way the MSV does it as this is a different faction.

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+  class rhsusf_usmc_marpat_wd_machinegunner: rhsusf_usmc_marpat_wd_autorifleman_m249 {// MMG
+    weapons[] = {WEST_MMG};
+    magazines[] = {WEST_MMG_MAG,WEST_PISTOL_MAG,WEST_BASE_GRENADES};
+    attachments[] = {};
+  };
+  class rhsusf_usmc_marpat_wd_machinegunner_ass: Fic_Spotter {// MMG Spotter/Ammo Bearer
+    vest[] = {"rhsusf_spc_light"};
+    backpackItems[] += {WEST_MMG_MAG2};
+  };
+  class rhsusf_usmc_lar_marpat_wd_machinegunner : Fic_Soldier_Carbine{// HMG
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: Will not have med supplies

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+    backpackItems[] = {};
+    backpack[] = { "RHS_M2_Gun_Bag" };
+  };
+  class rhsusf_usmc_lar_marpat_wd_marksman : Fic_Spotter{// HMG Spotter / Ammo carrier
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: Same as above

File: Loadouts/blu_usmc_m16_desert.hpp (hunk 5 above)
Comment (Alex6511) [inline Loadouts/blu_usmc_m16_desert.hpp]: His backpack is full, I'll be giving them some HEDP rounds in the next revision though.

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+  class rhsusf_usmc_lar_marpat_wd_combatcrewman : Fic_Spotter{// HAT Spotter
+    vest[] = {"rhsusf_spc_light"};
+    LAUNCHER_GEAR("B_Carryall_mcamo", WEST_HAT_MAG)
+  };
+  class rhsusf_usmc_marpat_wd_stinger : Fic_Soldier_Carbine {// SAM Gunner
+    vest[] = { "rhsusf_spc_light" };
+    LAUNCHER_GEAR("B_Carryall_mcamo", WEST_SAM_MAG)
+    launchers[] = {WEST_SAM};
+  };
+  class rhsusf_usmc_marpat_wd_grenadier: Fic_Soldier_Carbine {// Mortar Gunner
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: Just curious, why does he get a pistol?

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+  };
+  class Fic_eng: rhsusf_usmc_marpat_wd_crewman {
+    headgear[] = {"rhsusf_lwh_helmet_marpatd"};
+    items[] += {BASE_ENG};
+    backpackItems[] = {};
+  };
+  class rhsusf_usmc_marpat_wd_explosives: Fic_eng{// EOD Explodey guy
+    vest[] = { "rhsusf_spc_rifleman" };
+    //backpackItems[] += {BASE_EXP};    // uncomment to add explosives
+    //backpackItems[] += {BASE_MINE};   // uncomment to add mines
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: He should probably have one uncomment ed to start with.

File: Loadouts/blu_usmc_m16_desert.hpp (hunk 8 above)
Comment (Alex6511) [inline Loadouts/blu_usmc_m16_desert.hpp]: So he can sit on the mortar and driveby people obviously /s

Honestly there's not really a reason, I guess it can be removed.

File: Loadouts/blu_usmc_m16_desert.hpp (hunk 9 above)
Comment (Alex6511) [inline Loadouts/blu_usmc_m16_desert.hpp]: I'll make one default but I'm probably going to give explosives to the Assaultmen as well, in real life the marines rarely use dedicated EOD.

File: Loadouts/blu_usmc_m16_desert.hpp (hunk 3 above)
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: Fair enough. I know we've done simular substitutions in the past.

File: Loadouts/blu_usmc_m16_desert.hpp (hunk 5 above)
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: The gunner'so backpack will max, but one round will be taken out and put into the launcher, the script will then add the last rocket. For a total of one in the pipe 2 in the backpack.

Comment (Alex6511) [inline Loadouts/blu_usmc_m16_desert.hpp]: I actually just noticed that even the stuff defined here doesn't fit, so I'm going to have to make some other changes to the assaultman.

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+    TransportWeapons[] = {WEST_AT};
+    TransportMagazines[] = {WEST_RIFLE_MAG,WEST_RIFLE_MAG,WEST_CARBINE_MAG,WEST_AR_MAG,WEST_AR_MAG,WEST_GLRIFLE_MAG_HE};
+    TransportItems[] = {BASE_MEDICAL,BASE_MEDICAL,BASE_MEDICAL,BASE_MEDICAL};
+  };
+  class Helicopter {
+    TransportMagazines[] = {WEST_RIFLE_MAG,WEST_RIFLE_MAG,WEST_CARBINE_MAG,WEST_AR_MAG,WEST_AR_MAG,WEST_GLRIFLE_MAG_HE};
+    TransportItems[] = {BASE_MEDICAL,BASE_MEDICAL,BASE_MEDICAL,BASE_MEDICAL};
+  };
+  class Plane {};
+  class Ship_F {};
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: Remove these two empty classes

File: Loadouts/blu_usmc_m16_desert.hpp (hunk 5 above)
Comment (Alex6511) [inline Loadouts/blu_usmc_m16_desert.hpp]: It actually has a AA loaded already, with one in the backpack, in the current system.

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+  };
+  class rhsusf_usmc_marpat_wd_explosives: Fic_eng{// EOD Explodey guy
+    vest[] = { "rhsusf_spc_rifleman" };
+    //backpackItems[] += {BASE_EXP};    // uncomment to add explosives
+    //backpackItems[] += {BASE_MINE};   // uncomment to add mines
+    linkedItems[] = { WEST_BASE_LINKED };
+  };
+
+  class rhsusf_usmc_recon_marpat_wd_teamleader: rhsusf_usmc_marpat_wd_squadleader {// SF Squad Leader
+    //headgear[] = { "rhsusf_Bowman" };
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: Is this needed?

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+  };
+  class rhsusf_usmc_recon_marpat_wd_teamleader_fast: rhsusf_usmc_marpat_wd_teamleader{// SF FTL
+    //headgear[] = {"rhsusf_Bowman"};
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: Needed?

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 13]
```diff
@@ -0,0 +1,259 @@
...
+  class rhsusf_usmc_lar_marpat_wd_machinegunner : Fic_Soldier_Carbine{// HMG
+    backpackItems[] = {};
+    backpack[] = { "RHS_M2_Gun_Bag" };
+  };
+  class rhsusf_usmc_lar_marpat_wd_marksman : Fic_Spotter{// HMG Spotter / Ammo carrier
+    backpackItems[] = {};
+    backpack[] = {"RHS_M2_MiniTripod_Bag"};
+  };
+  class rhsusf_usmc_marpat_wd_smaw: Fic_Soldier_Carbine {// MAT Gunner
+    // Classnames for launcher set at the top of this file
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: Template standard, is this comment needed?

File: Loadouts/blu_usmc_m16_desert.hpp (hunk 12 above)
Comment (Alex6511) [inline Loadouts/blu_usmc_m16_desert.hpp]: Yes, gear for SF teams isn't finalized yet. Not an impeding issue.

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 14]
```diff
@@ -0,0 +1,259 @@
...
+  };
+  class rhsusf_usmc_recon_marpat_wd_marksman: rhsusf_usmc_marpat_wd_autorifleman_m249_ass{// SF AAR
+    weapons[] = {WEST_SNIPER};
+    magazines[] = {WEST_SNIPER_MAG, WEST_BASE_GRENADES};
+    attachments[] = {};
+  };
+  class rhsusf_usmc_recon_marpat_wd_rifleman_at: rhsusf_usmc_marpat_wd_riflemanat{// SF RAT
+  };
+
+
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: Remove two empty lines

File: Loadouts/blu_usmc_m16_desert.hpp (hunk 11 above)
Comment (Alex6511) [inline Loadouts/blu_usmc_m16_desert.hpp]: Yes, gear for SF teams isn't finalized yet. Not an impeding issue.

File: Loadouts/blu_usmc_m16_desert.hpp (hunk 10 above)
Comment (Alex6511) [inline Loadouts/blu_usmc_m16_desert.hpp]: https://github.com/BourbonWarfare/bwmf/blob/master/Loadouts/blu_us_m4_ocp.hpp#L55
Sure, but all of the other loadouts have that.

File: Loadouts/blu_usmc_m16_desert.hpp [hunk 15]
```diff
@@ -0,0 +1,259 @@
...
+  class fallback: rhsusf_usmc_marpat_wd_rifleman {}; //This means any faction member who doesn't match something will use this loadout
+};
+
+class rhs_faction_usn : rhs_faction_usmc_wd{
+  class rhsusf_navy_marpat_wd_medic : Fic_Soldier_Carbine{// SL Corpsman
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: Possible class scoping issue, has the medical been tested?

File: Loadouts/blu_usmc_m16_woodland.hpp [hunk 16]
```diff
@@ -0,0 +1,259 @@
+//Author: BWMF
```
Comment (AACO) [inline Loadouts/blu_usmc_m16_woodland.hpp]: Same quips as above file

File: f/pabstMarkers/defines_unitsAndGroups.sqf [hunk 17]
```diff
@@ -10,6 +10,16 @@
   ["UnitNATO_FSL_M","FM"], \
   ["UnitNATO_WSL_M","WSM"], \
   \
+	["unitMAR_1PLT_CORPS", "P1M"], \
```
Comment (AACO) [inline f/pabstMarkers/defines_unitsAndGroups.sqf]: None of the added times are up to spacing standards.

Comment (AACO) [inline f/pabstMarkers/defines_unitsAndGroups.sqf]: Also lower case u vs capital U every where elae.

File: Loadouts/blu_usmc_m16_desert.hpp (hunk 15 above)
Comment (Alex6511) [inline Loadouts/blu_usmc_m16_desert.hpp]: If you mean does the loadout properly apply with the medical gear, then yes, it works.

File: f/pabstMarkers/fn_getGroupMarkerStyle.sqf [hunk 18]
```diff
@@ -171,5 +171,10 @@ switch true do {
     _color = _pink;
     _size = [16,16];
   };
+    case (_input in ["REM"]): {
```
Comment (AACO) [inline f/pabstMarkers/fn_getGroupMarkerStyle.sqf]: Black med badge will probably just show up as a black box

Comment (Alex6511) [inline f/pabstMarkers/fn_getGroupMarkerStyle.sqf]: I'll just make it match the other FR icons which are orange.

File: CfgLoadouts.hpp (hunk 1 above)
Comment (AACO) [inline CfgLoadouts.hpp]: the file name should be:
usmc_us_m16_desert.hpp

the comment should be:
// US: M16 - FROG Desert

File: Loadouts/blu_usmc_m16_desert.hpp (hunk 11 above)
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: We should probably finalize the gear before a merge to master.

File: Loadouts/blu_usmc_m16_desert.hpp (hunk 15 above)
Comment (AACO) [inline Loadouts/blu_usmc_m16_desert.hpp]: Concern addressed then.

File: f/respawn/respawn_settings.sqf [hunk 19]
```diff
@@ -76,7 +80,20 @@ respawnMenuRoles = [
   ["C", "Crewman - Vehicle Crew"],
   ["D", "Crewman - Vehicle Driver"],
   ["PP", "Aircrewman - Pilot"],
-  ["PCC", "Aircrewman - Crew Chief (Repair)"]
+  ["PCC", "Aircrewman - Crew Chief (Repair)"],
+  ["VC","Vehicle Commander"],
+  ["VG","Vehicle Gunner"],
+  ["VD","Vehicle Driver"],
```
Comment (AACO) [inline f/respawn/respawn_settings.sqf]: Both vehicle gunner and vehicle driver are both listed above, please use those.

File: f/respawn/respawn_settings.sqf [hunk 20]
```diff
@@ -76,7 +80,20 @@ respawnMenuRoles = [
   ["C", "Crewman - Vehicle Crew"],
   ["D", "Crewman - Vehicle Driver"],
   ["PP", "Aircrewman - Pilot"],
-  ["PCC", "Aircrewman - Crew Chief (Repair)"]
+  ["PCC", "Aircrewman - Crew Chief (Repair)"],
+  ["VC","Vehicle Commander"],
```
Comment (AACO) [inline f/respawn/respawn_settings.sqf]: Backfill current factions to use the normal "Vehicle Crew" when this option is selected, so if someone choses this for any other factions it doesn't blow up.

File: f/respawn/respawn_settings.sqf [hunk 21]
```diff
@@ -76,7 +80,20 @@ respawnMenuRoles = [
...
+  ["VG","Vehicle Gunner"],
+  ["VD","Vehicle Driver"],
+  ["SSL","Special Squad Leader"],
+  ["SM","Special Corpsman"],
+  ["SFTL","Special Fireteam Leader"],
+  ["SAR","Special Autorifleman"],
+  ["SAAR","Special Ass Autorifleman"],
+  ["SRAT","Special Rifleman AT"],
```
Comment (AACO) [inline f/respawn/respawn_settings.sqf]: Do we want/need to respawn special teams?

File: f/respawn/respawn_settings.sqf [hunk 22]
```diff
@@ -85,7 +102,9 @@ respawnMenuFactions = [
   ["blu_f", "NATO"],
   ["opf_f", "CSAT"],
   ["ind_f", "AAF"],
-  ["rhs_faction_msv", "MSV"]
+  ["rhs_faction_msv", "MSV"],
+  ["rhs_faction_usmc_wd","USMC"],
+  ["rhs_faction_usn","USN - Medics"]
```
Comment (AACO) [inline f/respawn/respawn_settings.sqf]: "Medics" is way too generic, should be USN, or USN - Corpsmen

File: f/pabstMarkers/fn_getGroupMarkerStyle.sqf (hunk 18 above)
Comment (Alex6511) [inline f/pabstMarkers/fn_getGroupMarkerStyle.sqf]: Should be fixed now

File: Loadouts/usmc_us_m16_desert.hpp [hunk 23]
```diff
@@ -0,0 +1,255 @@
...
+    items[] += {BASE_ENG};
+    backpackItems[] = {};
+  };
+  class rhsusf_usmc_marpat_wd_explosives: Fic_eng{// EOD Explodey guy
+    vest[] = { "rhsusf_spc_rifleman" };
+    backpackItems[] += {BASE_EXP};    // uncomment to add explosives
+    //backpackItems[] += {BASE_MINE};   // uncomment to add mines
+    linkedItems[] = { WEST_BASE_LINKED };
+  };
+
```
Comment (AACO) [inline Loadouts/usmc_us_m16_desert.hpp]: Remove newline

File: Loadouts/usmc_us_m16_desert.hpp [hunk 24]
```diff
@@ -0,0 +1,255 @@
...
+  };
+  class rhsusf_usmc_recon_marpat_wd_marksman: rhsusf_usmc_marpat_wd_autorifleman_m249_ass{// SF AAR
+    weapons[] = {WEST_SNIPER};
+    magazines[] = {WEST_SNIPER_MAG, WEST_BASE_GRENADES};
+    attachments[] = {};
+  };
+  class rhsusf_usmc_recon_marpat_wd_rifleman_at: rhsusf_usmc_marpat_wd_riflemanat{// SF RAT
+  };
+
+
```
Comment (AACO) [inline Loadouts/usmc_us_m16_desert.hpp]: Remove newlines

File: Loadouts/usmc_us_m16_woodland.hpp [hunk 25]
```diff
@@ -0,0 +1,261 @@
...
+    items[] += {BASE_ENG};
+    backpackItems[] = {};
+  };
+  class rhsusf_usmc_marpat_wd_explosives: Fic_eng{// EOD Explodey guy
+    vest[] = { "rhsusf_spc_rifleman" };
+    backpackItems[] += {BASE_EXP};    // uncomment to add explosives
+    //backpackItems[] += {BASE_MINE};   // uncomment to add mines
+    linkedItems[] = { WEST_BASE_LINKED };
+  };
+
```
Comment (AACO) [inline Loadouts/usmc_us_m16_woodland.hpp]: remove new line

File: Loadouts/usmc_us_m16_woodland.hpp [hunk 26]
```diff
@@ -0,0 +1,261 @@
...
+    uniform[] = {"rhs_uniform_FROG01_m81"};
+    weapons[] = {WEST_SNIPER};
+    magazines[] = {WEST_SNIPER_MAG, WEST_BASE_GRENADES};
+    attachments[] = {};
+  };
+  class rhsusf_usmc_recon_marpat_wd_rifleman_at: rhsusf_usmc_marpat_wd_riflemanat{// SF RAT
+  uniform[] = {"rhs_uniform_FROG01_m81"};
+  };
+
+
```
Comment (AACO) [inline Loadouts/usmc_us_m16_woodland.hpp]: Remove newlines
